src/accelerate/utils/launch.py

```diff
@@ -128,7 +128,8 @@ def prepare_multi_gpu_env(args: argparse.Namespace) -> Dict[str, str]:
     if main_process_port is None:
         main_process_port = 29500
 
-    if is_port_in_use(main_process_port):
+    need_port_check = num_machines <= 1 or int(args.machine_rank) == 0
+    if need_port_check and is_port_in_use(main_process_port):
```

Comment:
Just a small nit though thinking on this more, can we add a comment here just mentioning how/why we should only check on the main node for this, in case others want to learn why as well :)

Reply:
cc @Jingru :)

Reply:
> cc @Jingru :)

already add some comments